be/src/exprs/agg/group_concat.h

```diff
@@ -292,4 +299,422 @@ class GroupConcatAggregateFunction
+        : public AggregateFunctionBatchHelper<GroupConcatAggregateStateV2, GroupConcatAggregateFunctionV2> {
+public:
+    // group_concat(a, b order by c, d), the arguments are a,b,',',c,d
+    void create_impl(FunctionContext* ctx, GroupConcatAggregateStateV2& state) const {
+        auto num = ctx->get_num_args();
+        state.data_columns = new Columns;
+        auto order_by_num = ctx->get_nulls_first().size();
+        state.output_col_num = num - order_by_num - 1; // excluding separator column
+        if (UNLIKELY(state.output_col_num <= 0)) {
+            ctx->set_error("group_concat output column should not be empty", false);
+            return;
+        }
+        for (auto i = 0; i < state.output_col_num; ++i) {
+            if (UNLIKELY(!is_string_type(ctx->get_arg_type(i)->type))) {
+                ctx->set_error(std::string(std::to_string(i) + "-th input of group_concat is not string type, but is " +
+                                           type_to_string(ctx->get_arg_type(i)->type))
+                                       .c_str(),
+                               false);
+                return;
+            }
+        }
+        for (auto i = 0; i < num; ++i) {
+            state.data_columns->emplace_back(ctx->create_column(*ctx->get_arg_type(i), true));
+        }
+        DCHECK(ctx->get_is_asc_order().size() == ctx->get_nulls_first().size());
+    }
+
+    void reset(FunctionContext* ctx, const Columns& args, AggDataPtr __restrict state) const override {
+        auto& state_impl = this->data(state);
+        if (state_impl.data_columns != nullptr) {
+            for (auto& col : *state_impl.data_columns) {
+                col->resize(0);
+            }
+        }
+    }
+
+    // reject null for output columns, but non-output columns may be null
+    void update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state,
+                size_t row_num) const override {
+        auto num = ctx->get_num_args();
+        GroupConcatAggregateStateV2& state_impl = this->data(state);
+        if (state_impl.data_columns == nullptr) {
+            create_impl(ctx, state_impl);
+        }
+        for (auto i = 0; i < state_impl.output_col_num; ++i) {
+            if (columns[i]->is_nullable() && columns[i]->is_null(row_num)) {
+                return;
+            }
+        }
+
+        for (auto i = 0; i < num; ++i) {
+            // non-output columns is null
+            if (i >= state_impl.output_col_num && (columns[i]->is_nullable() && columns[i]->is_null(row_num))) {
+                this->data(state).update_nulls(ctx, i, 1);
+                continue;
+            }
+            auto* data_col = columns[i];
+            auto tmp_row_num = row_num;
+            if (columns[i]->is_constant()) {
+                // just copy the first const value.
+                data_col = down_cast<const ConstColumn*>(columns[i])->data_column().get();
+                tmp_row_num = 0;
+            }
```

Comment:
Missing branch for processing NullableColumn

Reply:
nullable column can be update in the state, as the data columns in state are nullable.